DEVELOPING_A_RUNTIME.md

```diff
@@ -16,146 +36,161 @@ A **required** exported constant that decides which version of the Runtime API t
 
 The latest and suggested version is `3`.
 
-### `analyze`
+**Example:**
 
-An **optional** exported function that returns a unique fingerprint used for the purpose of [build de-duplication](https://vercel.com/docs/v2/platform/deployments#deduplication). If the `analyze` function is not supplied, a random fingerprint is assigned to each build.
-
-```js
-export analyze({
-  files: Files,
-  entrypoint: String,
-  workPath: String,
-  config: Object
-}) : String fingerprint
+```typescript
+export const version = 3;
 ```
 
-If you are using TypeScript, you should use the following types:
+### `build()`
 
-```ts
-import { AnalyzeOptions } from '@vercel/build-utils'
+A **required** exported function that returns a [Serverless Function](#serverless-function).
 
-export analyze(options: AnalyzeOptions) {
-  return 'fingerprint goes here'
-}
-```
+What's a Serverless Function? Read about [Serverless Functions](https://vercel.com/docs/v2/serverless-functions/introduction) to learn more.
 
-### `build`
+**Example:**
 
-A **required** exported function that returns a [Serverless Function](#serverless-function).
+```typescript
+import { BuildOptions, createLambda } from '@vercel/build-utils';
 
-What's a Serverless Function? Read about [Serverless Functions](https://vercel.com/docs/v2/serverless-functions/introduction) to learn more.
+export async function build(options: BuildOptions) {
+  // Build the code here…
 
-```js
-build({
-  files: Files,
-  entrypoint: String,
-  workPath: String,
-  config: Object,
-  meta?: {
-    isDev?: Boolean,
-    requestPath?: String,
-    filesChanged?: Array<String>,
-    filesRemoved?: Array<String>
-  }
-}) : {
-  watch?: Array<String>,
-  output: Lambda,
-  routes?: Object
+  const lambda = createLambda(/* … */);
+  return {
+    output: lambda,
+    watch: [
+      // Dependent files to trigger a rebuild in `vercel dev` go here…
```

Comment:
Is this still relevant when using `startDevServer()`?

Reply:
It's not, but you might not implement `startDevServer()`. Eventually though, ya, this will probably be phased out.

We could add a note saying that if `startDevServer()` is implemented then this array is not used. Though the `startDevServer()` documentation mentions how if it's implemented then `build()` is not invoked at all, so this part is kind of implied.